source/common/upstream/subset_lb.cc

```diff
@@ -544,10 +544,18 @@ void SubsetLoadBalancer::HostSubsetImpl::update(const HostVector& hosts_added,
   }
 }
 
-HostSetImplPtr SubsetLoadBalancer::PrioritySubsetImpl::createHostSet(uint32_t priority) {
+HostSetImplPtr SubsetLoadBalancer::PrioritySubsetImpl::createHostSet(
+    uint32_t priority, absl::optional<uint32_t> overprovisioning_factor) {
+  // Use original hostset's overprovisioning_factor.
   RELEASE_ASSERT(priority < original_priority_set_.hostSetsPerPriority().size(), "");
-  return HostSetImplPtr{new HostSubsetImpl(*original_priority_set_.hostSetsPerPriority()[priority],
-                                           locality_weight_aware_)};
+
+  const HostSetPtr& host_set = original_priority_set_.hostSetsPerPriority()[priority];
+
+  UNREFERENCED_PARAMETER(overprovisioning_factor);
```

Comment:
I think we can remove this viten the new if/ASSERT below.
Alternately you can rewrite the assert so the if is compiled out as well
ASSERT(!overprovisioning_factor.has_value() || overprovisioning_factor.value() == host_set->overprovisioning_factor())

Reply:
Done.